state/accountsDB.go

```diff
@@ -1027,22 +1027,48 @@ func (adb *AccountsDB) SnapshotState(rootHash []byte) {
 	log.Trace("accountsDB.SnapshotState", "root hash", rootHash)
 	trieStorageManager.EnterPruningBufferingMode()
 
+	wg := &sync.WaitGroup{}
```

Comment:
L1030-L1035 could have been wrapped in a `newSnapshotStatistics` unexported constructor
That will avoid the duplicated code and required mental mapping for L1109

Reply:
Done.